### ISSUE TITLE
Support annotationProcessorPaths for the Graph Generator

### ISSUE THREAD

**Issue by jacob-swanson** [Cosium/spring-data-jpa-entity-graph#53]
**What steps will reproduce the problem?**
1. Add spring-data-jpa-entity-graph dependency
2. Add spring-data-jpa-entity-graph-generator in annotationProcessorPaths
3. `mvn compile`

Example maven-complier-plugin configuration:
```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <configuration>
        <source>${java.version}</source>
        <target>${java.version}</target>
        <annotationProcessorPaths>
            <path>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-configuration-processor</artifactId>
                <version>${spring-boot.version}</version>
            </path>
            <path>
                <groupId>org.hibernate</groupId>
                <artifactId>hibernate-jpamodelgen</artifactId>
                <version>${hibernate.version}</version>
            </path>
            <path>
                <groupId>com.cosium.spring.data</groupId>
                <artifactId>spring-data-jpa-entity-graph-generator</artifactId>
                <version>2.4.1</version>
            </path>
        </annotationProcessorPaths>
        <compilerArgs>
            <compilerArg>-Amapstruct.suppressGeneratorTimestamp=true</compilerArg>
            <compilerArg>-Amapstruct.suppressGeneratorVersionInfoComment=true</compilerArg>
        </compilerArgs>
    </configuration>
</plugin>
```

**What is the expected output?**
Build success

**What happens instead?**
Build fails with: `java.lang.NoClassDefFoundError: javax/persistence/metamodel/StaticMetamodel: javax.persistence.metamodel.StaticMetamodel`

**Environment:**
 - Spring Data JPA version (not the Spring Boot one): 2.4.2
 - ORM with version: 5.4.25.Final
 - spring-data-jpa-entity-graph version: 2.4.1

**Additional context**
To fix the issue I tested removing `<scope>provided</scope>` from generator/pom.xml.
Afterward, the build partially succeeded, but this failure is due to a different problem that shouldn't be directly related to this issue.

**Comment by jacob-swanson** [comment]
@reda-alaoui yes, the issue is fixed. Thank you.

**Comment by austalakov** [comment]
Is there a way to have this fix for 2.3.x as well? We are using Spring Data JPA 2.3.3, so have to use spring-data-jpa-entity-graph 2.3.1

**Comment by reda-alaoui** [comment]
@austalakov I just cherry picked it on 2.3.x and released version 2.3.2.
You should be able to use it in a few minutes.
